platform/posix/transport/utest/plaintext_utest.c

```diff
@@ -190,20 +222,62 @@ void test_Plaintext_Send_Network_Error( void )
 
     for( i = 0; i < sizeof( errorNumbers ); i++ )
     {
+        getsockopt_ExpectAnyArgsAndReturn( 0 );
+        select_ExpectAnyArgsAndReturn( 1 );
         send_ExpectAnyArgsAndReturn( SEND_RECV_ERROR );
         errno = errorNumbers[ i ];
         bytesSent = Plaintext_Send( &networkContext,
                                     plaintextBuffer,
                                     BYTES_TO_SEND );
 
-        /* EAGAIN / EWOULDBLOCK imply no data to send. */
-        if( ( errno == EAGAIN ) || ( errno == EWOULDBLOCK ) )
-        {
-            TEST_ASSERT_EQUAL( 0, bytesSent );
-        }
-        else
-        {
-            TEST_ASSERT_EQUAL( SEND_RECV_ERROR, bytesSent );
-        }
+        TEST_ASSERT_EQUAL( SEND_RECV_ERROR, bytesSent );
     }
 }
+
+/**
+ * @brief Test that #Plaintext_Send returns an error when #send receives
+ * zero bytes from #send implying that the peer has closed the connection.
+ */
+void test_Plaintext_Send_Zero_Bytes_Received( void )
+{
+    int32_t bytesSent;
+
+    getsockopt_ExpectAnyArgsAndReturn( -1 );
+    select_ExpectAnyArgsAndReturn( 1 );
+    send_ExpectAnyArgsAndReturn( 0 );
+    bytesSent = Plaintext_Send( &networkContext,
+                                plaintextBuffer,
+                                BYTES_TO_SEND );
+    TEST_ASSERT_EQUAL( SEND_RECV_ERROR, bytesSent );
+}
+
+/**
+ * @brief Test that #Plaintext_Send returns 0 bytes when the socket has no activity.
```

Comment:
This can instead be called the send timeout test

Reply:
Lovely